### PR TITLE
RDK-6707 - Wait on reconfig workers in tests

### PR DESCRIPTION
**Pull request by cheukt**
ticket is https://viam.atlassian.net/browse/RSDK-6707

basically the failure case is that we succeed the test but cleanup before the reconfigure workers are fully cleaned up, which leads to a race where the worker tries to log but the logger is already `done`.

I went ahead and moved the local_robot_test.go into the robotimpl package (no issues surprisingly), which meant I can just use all the same helper and wait for the reconfigWorkers within. let me know if I shouldn't do that

### CODE REVIEW

File: robot/impl/utils.go [hunk 1]
```diff
@@ -27,6 +23,8 @@ func setupLocalRobot(
 	test.That(t, err, test.ShouldBeNil)
 	t.Cleanup(func() {
 		test.That(t, r.Close(ctx), test.ShouldBeNil)
+		// Wait for reconfigureWorkers here because localRobot.Close does not.
+		r.(*localRobot).reconfigureWorkers.Wait()
```
Comment (benjirewis) [inline robot/impl/utils.go:27]: Might as well check your type assertion here with a `, ok`.

File: robot/impl/local_robot_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package robotimpl_test
+package robotimpl
```
Comment (benjirewis) [inline robot/impl/local_robot_test.go:1]: Seems fine; so `robot_reconfigure_test.go` and `local_robot_test.go` are now just separate files testing basically the same logic? With hypothetically more focus on the reconfiguration side of things for the former?

**Review by benjirewis — APPROVED** [review]
Pretty much LGTM just a question and a nit.

File: robot/impl/local_robot_test.go (hunk 2 above)
Comment (cheukt) [reply robot/impl/local_robot_test.go:1]: I would argue they've always been testing very similar things, but yes. I'd like to at some point soon go into these tests and decouple them from the rest of the RDK as well and consolidate the tests somewhat (https://viam.atlassian.net/browse/RSDK-2670)

**Review by dgottlieb — APPROVED** [review]

**Pull request merged**